### PR TITLE
feat: star routing gateway

### PR DESCRIPTION
**Pull request by JoanFM**

### CODE REVIEW

File: jina/peapods/stream/gateway.py [hunk 1]
```diff
@@ -1,20 +1,86 @@
...
+            )
+            if len(filtered_partial_responses) > 1:
+                docs = DocumentArray(
+                    [
+                        d
+                        for r in filtered_partial_responses
+                        for d in getattr(r.request, 'docs')
+                    ]
+                )
+                groundtruths = DocumentArray(
```
Comment (jacobowitz) [inline jina/peapods/stream/gateway.py:68]: we didnt auto merge the groundtruths before, did we? (only in cases where we had some executor I think?)

File: jina/peapods/stream/client.py [hunk 2]
```diff
@@ -73,6 +80,40 @@ async def _receive(self):
                     future.cancel()
                 self.request_buffer.clear()
 
+    def _handle_request(self, request: 'Request') -> 'asyncio.Future':
+        """
+        For zmq & grpc data requests from gateway, for each request in the iterator, we send the `Message`
```
Comment (jacobowitz) [inline jina/peapods/stream/client.py:85]: there is no zmq anymore

**Review by jacobowitz — CHANGES_REQUESTED** [review]
Looks good! 
One improvement could be made in the BaseStreamer I think: We dont need the result_queue/callback mechanism anymore, because the sending Future from the new Graph will hold the result and can be used to retrieve it and yielding it to the client

File: jina/peapods/stream/client.py (hunk 2 above)
Comment (JoanFM) [reply jina/peapods/stream/client.py:85]: I will review docstrings later

File: jina/peapods/stream/gateway.py (hunk 1 above)
Comment (JoanFM) [reply jina/peapods/stream/gateway.py:68]: yes, which seems a little bit dangerous, We can revisit this later

**Pull request merged**